examples/api/lib/widgets/navigator_pop_handler/navigator_pop_handler.0.dart

```diff
@@ -16,10 +16,35 @@ class NavigatorPopHandlerApp extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
+      restorationScopeId: 'root',
       initialRoute: '/',
-      routes: <String, WidgetBuilder>{
-        '/': (BuildContext context) => _HomePage(),
-        '/nested_navigators': (BuildContext context) => const NestedNavigatorsPage(),
+      onGenerateRoute: (RouteSettings settings) {
```

Comment:
why convert the onGenerateRoute? it seems to make it slightly more verbose..

Reply:
I need to be able to provide RouteSettings.name.